sycl/include/sycl/ext/oneapi/properties/property.hpp

```diff
@@ -0,0 +1,174 @@
+//==---------- properties.hpp --- SYCL extension property tooling ----------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// HOW-TO: Add new compile-time property
+//  1. Add a new enumerator to
+//     `sycl::ext::oneapi::experimental::detail::PropKind` representing the new
+//     property. Increment
+//     `sycl::ext::oneapi::experimental::detail::PropKind::PropKindSize`
+//  2. Define property key class with `value_t` that must be `property_value`
+//     with the first template argument being the property class itself. The
+//     name of the key class must be the property name suffixed by `_key`, i.e.
+//     for a property `foo` the class should be named `foo_key`.
+//  3. Add an `inline constexpr` variable in the same namespace as the property
+//     key. The variable should have the same type as `value_t` of the property
+//     class, e.g. for a property `foo`, there should be a definition
+//     `inline constexpr foo_key::value_t foo`.
+//  4. Specialize `sycl::ext::oneapi::experimental::is_property` and
+//     `sycl::ext::oneapi::experimental::is_property_of` for the property key
```

Comment:
do we specialize `is_property` for the key? Or do we specialize `is_property_key` ?

Same applies to `is_property_of` vs `is_property_key_of` ?

Reply:
Good eye. This was missed with the renaming and should be `is_property_key` and `is_property_key_of`.